openexr: 2.4.1 -> 2.5.2

r-ryantm opened the pull request:
Semi-automatic update generated by [nixpkgs-update](https://github.com/ryantm/nixpkgs-update) tools. This update was made based on information from https://github.com/AcademySoftwareFoundation/openexr/releases.



meta.description for openexr is: "A high dynamic-range (HDR) image file format"

meta.homepage for openexr is: "https://www.openexr.com/"

###### Updates performed
- Version update

###### To inspect upstream changes


- Releases on GitHub
    - https://github.com/AcademySoftwareFoundation/openexr/releases/tag/v2.5.1
    - https://github.com/AcademySoftwareFoundation/openexr/releases/tag/v2.5.0
- [Compare changes on GitHub](https://github.com/AcademySoftwareFoundation/openexr/compare/v2.4.1...v2.5.1)


###### Impact

<details>
<summary>
<b>Checks done</b> (click to expand)
</summary>

---

- built on NixOS
- The tests defined in `passthru.tests`, if any, passed

- 0 of 0 passed binary check by having a zero exit code.
- 0 of 0 passed binary check by having the new version present in output.
- found 2.5.1 with grep in /nix/store/20kip7s93fb8f1yg2z552594fcj9h5fy-openexr-2.5.1-bin
- directory tree listing: https://gist.github.com/c001bea67fe0c4e69f15ebc99ca8e0d9

- du listing: https://gist.github.com/614309a12ead03373b3f872d99b2b08f


---

</details>
<details>
<summary>
<b>Rebuild report</b> (if merged into master) (click to expand)
</summary>

```
2903 total rebuild path(s)

949 package rebuild(s)

949 x86_64-linux rebuild(s)
838 i686-linux rebuild(s)
287 x86_64-darwin rebuild(s)
829 aarch64-linux rebuild(s)


First fifty rebuilds by attrpath
R
_90secondportraits
abiword
adapta-gtk-theme
aeon
aesop
alembic
almanah
anbox
apostrophe
appleseed
aptly
aravis
arc-icon-theme
arc-theme
ardour
areca
arx-libertatis
asc-key-to-qr-code-gif
asciidoc-full
asciidoc-full-with-plugins
astroid
asymptote
audacious
audaciousQt5
audio-recorder
autotrace
awesome
awesome-4-0
azureus
balsa
bareos
baresip
beets
betterlockscreen
bibata-cursors
bibata-cursors-translucent
bibata-extra-cursors
birdfont
blender
blockhash
bombono
bookworm
brasero
brasero-original
btrbk
byzanz
caffe
caffe2
calibre
```

</details>

<details>
<summary>
<b>Instructions to test this update</b> (click to expand)
</summary>

---

Either **download from Cachix**:
```
nix-store -r /nix/store/20kip7s93fb8f1yg2z552594fcj9h5fy-openexr-2.5.1-bin \
  --option binary-caches 'https://cache.nixos.org/ https://r-ryantm.cachix.org/' \
  --option trusted-public-keys '
  r-ryantm.cachix.org-1:gkUbLkouDAyvBdpBX0JOdIiD2/DP1ldF3Z3Y6Gqcc4c=
  cache.nixos.org-1:6NCHdD59X431o0gWypbMrAURkbJ16ZPMQFGspcDShjY=
  '
```
(r-ryantm's Cachix cache is only trusted for this store-path realization.)
For the Cachix download to work, your user must be in the `trusted-users` list or you can use `sudo` since root is effectively trusted.

Or, **build yourself**:
```
nix-build -A openexr https://github.com/r-ryantm/nixpkgs/archive/5e4e4a82c48fa3a17b9cfb7d41468e74af180618.tar.gz
```

After you've downloaded or built it, look at the files and if there are any, run the binaries:
```
ls -la /nix/store/20kip7s93fb8f1yg2z552594fcj9h5fy-openexr-2.5.1-bin
ls -la /nix/store/20kip7s93fb8f1yg2z552594fcj9h5fy-openexr-2.5.1-bin/bin
```

---

</details>
<br/>



### Pre-merge build results

NixPkgs review skipped

---

###### Maintainer pings

veprbl commented:
@GrahamcOfBorg eval

veprbl commented:
@GrahamcOfBorg eval

jtojnar reviewed (dismissed):
Looks like there has been an regression in pkg-config files: https://github.com/AcademySoftwareFoundation/openexr/commit/0b26a9dedda4924841323677f1ce0bce37bfbeb4

We will need to fix them like here: https://github.com/fmtlib/fmt/pull/1657

FRidh commented:
@GrahamcOfBorg eval

jtojnar commented:
Finally opened https://github.com/AcademySoftwareFoundation/openexr/pull/815

jtojnar committed:
openexr: 2.4.1 -> 2.5.2

Needed to apply patch to fix pkg-config files so also removed patch that is unused since the switch to CMake during patches refactoring.

jtojnar commented:
cc @glittershark @eraserhd @flokli re #86241
cc @nh2 re #94205

glittershark reviewed (approved):
Tested build and execution of all binaries on darwin and everything appears to work. Code also LGTM.

nh2 reviewed (commented):
Testing this on musl practically requires #95234 because otherwise `util-linux` (a dependency) does not build.

I'm currently building it on top of that.

nh2 commented:
The change to the musl patch seems to not work; `pkgsMusl.ilmbase` fails with:

```
applying patch /nix/store/g6ns8r8f7s8cx51asi404fa2wny9sy30-ilmbase-musl-_fpstate.patch.patch
can't find file to patch at input line 3
Perhaps you used the wrong -p or --strip option?
The text leading up to this was:
--------------------------
|--- IlmBase/IexMath/IexMathFpu.cpp.orig        2019-12-01 15:42:08.143387128 +0100
|+++ IlmBase/IexMath/IexMathFpu.cpp     2019-12-01 15:43:02.402389927 +0100
--------------------------
File to patch: 
Skip this patch? [y] 
Skipping patch.
1 out of 1 hunk ignored
builder for '/nix/store/vsrbwdxfyhsbcvvbfjqz1qhkp2710z31-ilmbase-2.5.2.drv' failed with exit code 1
```

nh2 commented on pkgs/development/libraries/openexr/default.nix:
```diff
@@ -9,49 +9,40 @@
-    if (stdenv.hostPlatform.isLinux && stdenv.hostPlatform.libc != "glibc")
-      then
-        ''
-          patch -p0 < ${non_glibc_fpstate_patch}
```
I don't think this can be removed, because the patch looks like

```
--- IlmBase/IexMath/IexMathFpu.cpp.orig
+++ IlmBase/IexMath/IexMathFpu.cpp
```

not starting with `a/IlmBase`, so `-p0` is required, and `patches` uses `-p1`.

nh2 replied:
This fixes it:

```diff
-      sha256 = "0appzbs9pd6dia5pzxmrs9ww35shlxi329ks6lchwzw4f2a81arz";
+      sha256 = "1bmyg4qfbz2p5iflrakbj8jzs85s1cf4cpfyclycnnqqi45j8m8d";
+      # The patch's files are written as `IlmBase/...`, this turns it into
+      # `a/IlmBase/...`, so that the `patch -p1` that `patches` does works.
+      extraPrefix = ""; # Changing this requires changing the `sha256` (fixed-output)!
```

It works because of https://github.com/NixOS/nixpkgs/blob/07f111eac53bd351dc8e509751e1a2f5d38111ae/pkgs/build-support/fetchpatch/default.nix#L31-L34

which I find confusing; you'd think `extraPrefix = ""` doesn't do anything, but it puts an `a/` prefix in.

But I'll force-push that fix into this PR.

nh2 commented:
> But I'll force-push that fix into this PR.

Done.

nh2 reviewed (approved):
With my force-pushed change, `pkgsMusl.ilmbase` and `pkgsMusl.openexr` build on top of #95234.
